src/transformers/modeling_utils.py

```diff
@@ -833,21 +852,40 @@ def _generate_no_beam_search(
                 next_token = torch.argmax(next_token_logits, dim=-1)
 
             # update generations and finished sentences
-            tokens_to_add = next_token * unfinished_sents + pad_token_id * (1 - unfinished_sents)
+            if eos_token_ids is not None:
```

Comment:
1) I'd change this comment to "pad finished sentences".
2) Since we know `pad_token_id` is not None, do we still need to do different things depending on eos_token_ids?

Reply:
1. yes!
2. If eos_token_ids is None, then pad_token_id can also be None so that sentences should never be padded and the line `tokens_to_add = next_token` should be used.